core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -218,4 +218,13 @@ private[spark] object UI {
     .stringConf
     .transform(_.toUpperCase(Locale.ROOT))
     .createWithDefault("LOCAL")
+
+  val UI_THREADS = ConfigBuilder("spark.ui.threads")
```

Comment:
I would really rather not add yet another config. Can Jetty thread pools simply expand as needed up to some large max? that kind of resolves this automatically.

Reply:
> Can Jetty thread pools simply expand as needed up to some large max? 

Yes. And currently, we are using the default value of 200 for all cases. Adding this config to make max no. to be configurable.

For example, for a single Spark application with UI enabled, we can limit it to a small value. This can be a protection of the application itself when `http://localhost:4040/api/v1` is requested too often.

Reply:
Is there any reason to limit it, if threads are only used if needed? just increase the max?

Reply:
When bottleneck hits and it's not caused by not enough threads.

And similarly, we have limited the size of acceptors https://github.com/apache/spark/pull/32539/files#diff-bb2ed90c22fe4b314488f3ffbd32e9998a9fd2dae994ea575b264d81f060cba7R299-R300.

Reply:
On the other hand, the default 200 might be easy to hit 
![image](https://user-images.githubusercontent.com/8326978/118219985-294a2900-b4ad-11eb-876b-0928baa8ea58.png)

Reply:
I don't quite understand - are you saying 200 isn't enough? then increase the max.
Is 200 too many and you have too many threads lingering? then can we make the scale-down any more aggressive?
But in any event what is the problem?

Reply:
History Server / Spark Standalone Master Web / Spark Application Live UI, I didn't mean that 200 is enough or not for a particular case but not flexible enough for these different scenarios.

Reply:
Why not just up the max, if it autoscales?
The problem is, you can expose this config, but, it only helps if users find it and set it. That's unlikely for most users. If we can simply make it work with a default change, that's much better.

Reply:
> Why not just up the max, if it autoscales?

Yes, it autoscales, but I don't the max no. just go up but in some cases, I don't want it to go that high, maybe just a few threads to meet the basic need of jetty server.

Reply:
>  In some cases, I don't want it to go that high, maybe just a few threads to meet the basic need of a jetty server.

For example, for a machine that used to launch Spark applications, we can limit this size to reduce to risk of getting `java.lang.OutOfMemoryError: Unable to create new native thread `

Reply:
I get it. You may need to serve 200 concurrent requests, but want to limit it to 50 to limit the number of threads. Thread stacks are not that big. Would you not rather up the memory a little? that is an easy an obvious lever users can already pull to resolve this without limiting concurrency.

Reply:
It is one example not a case I am facing. Increasing mem can solve the issue in the example I gave. But for a overbooking or heavy app scheduler node，thread or memory are limit.

On the other hand，a HISTORY SERVER might need larger thread no. to meet high concurrency

Reply:
It's probably much better and easier to increase memory than have to figure out a separate thread configuration, which is quite secondary.  I am still not super clear on whether you mean this would generally be used to increase or decrease the default.

Reply:
Something like what we do in Spark thrift server, 
 we have an up limit [maxWorkerThreads ](https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java#L169) for a auto-scalable thread pool [HiveServer2-HttpHandler-Pool](https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/thrift/ThriftHttpCLIService.java#L66)